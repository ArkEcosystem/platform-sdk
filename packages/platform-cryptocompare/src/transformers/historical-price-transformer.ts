import { Contracts, Utils } from "@arkecosystem/platform-sdk";

export class HistoricalPriceTransformer implements Contracts.HistoricalTransformer {
	public constructor(private readonly data: Contracts.KeyValuePair) {}

	public transform(options: Contracts.KeyValuePair): Contracts.HistoricalData {
		const datasets = this.data.map((value) => value.close);

		return {
			labels: this.data.map((value) => Utils.dayjs(value.time * 1000).format(options.dateFormat)),
			datasets,
			min: Math.min(...datasets),
			max: Math.max(...datasets),
		};
	}
}
