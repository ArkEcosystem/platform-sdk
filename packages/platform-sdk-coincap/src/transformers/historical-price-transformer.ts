import { Contracts } from "@arkecosystem/platform-sdk";
import { DateTime } from "@arkecosystem/platform-sdk-intl";

import { convertToCurrency } from "../utils";

export class HistoricalPriceTransformer implements Contracts.HistoricalTransformer {
	public constructor(private readonly data: Contracts.KeyValuePair) {}

	public transform(options: Contracts.KeyValuePair): Contracts.HistoricalData {
		const { token, currency, rates, dateFormat } = options;

		const tokenId = token.toUpperCase();
		const datasets = {};

		for (const value of Object.values(this.data) as any) {
			datasets[DateTime.make(value.date).format(dateFormat)] = convertToCurrency(value.priceUsd, {
				from: currency,
				to: tokenId,
				base: tokenId,
				rates,
			});
		}

		const datasetValues: number[] = Object.values(datasets);

		return {
			labels: Object.keys(datasets),
			datasets: datasetValues,
			min: Math.min(...datasetValues),
			max: Math.max(...datasetValues),
		};
	}
}
