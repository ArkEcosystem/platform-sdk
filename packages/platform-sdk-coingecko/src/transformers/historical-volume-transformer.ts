import { Contracts } from "@arkecosystem/platform-sdk";
import { DateTime } from "@arkecosystem/platform-sdk-intl";

export class HistoricalVolumeTransformer implements Contracts.HistoricalTransformer {
	public constructor(private readonly data: Contracts.KeyValuePair) {}

	public transform(options: Contracts.KeyValuePair): Contracts.HistoricalData {
		const datasets = {};

		for (let i = 0; i < this.data.total_volumes.length; i += 24) {
			datasets[this.data.total_volumes[i][0]] = this.data.total_volumes[i][1];
		}

		const datasetValues: number[] = Object.values(datasets);

		return {
			labels: Object.keys(datasets).map((time) => DateTime.make(time).format(options.dateFormat)),
			datasets: datasetValues,
			min: Math.min(...datasetValues),
			max: Math.max(...datasetValues),
		};
	}
}
