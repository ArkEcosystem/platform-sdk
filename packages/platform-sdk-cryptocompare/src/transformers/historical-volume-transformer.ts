import { Contracts, Utils } from "@arkecosystem/platform-sdk";

export class HistoricalVolumeTransformer implements Contracts.HistoricalTransformer {
	public constructor(private readonly data: Contracts.KeyValuePair) {}

	public transform(options: Contracts.KeyValuePair): Contracts.HistoricalData {
		const datasets = this.data.map((value) => value.volumeto);

		return {
			labels: this.data.map((value) => Utils.dayjs(value.time * 1000).format(options.dateFormat)),
			datasets,
			min: Math.min(...datasets),
			max: Math.max(...datasets),
		};
	}
}
