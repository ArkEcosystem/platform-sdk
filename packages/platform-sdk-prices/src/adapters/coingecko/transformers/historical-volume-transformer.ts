import { Contracts, Utils } from "@arkecosystem/platform-sdk";

import { HistoricalData, HistoricalTransformer } from "../../../contracts/historical";

export class HistoricalVolumeTransformer implements HistoricalTransformer {
	public constructor(private readonly data: Contracts.KeyValuePair) {}

	public transform(options: Contracts.KeyValuePair): HistoricalData {
		const datasets = {};

		for (let i = 0; i < this.data.total_volumes.length; i += 24) {
			datasets[this.data.total_volumes[i][0]] = this.data.total_volumes[i][1];
		}

		const datasetValues: number[] = Object.values(datasets);

		return {
			labels: Object.keys(datasets).map((time) => Utils.dayjs(time).format(options.dateFormat)),
			datasets: datasetValues,
			min: Math.min(...datasetValues),
			max: Math.max(...datasetValues),
		};
	}
}
