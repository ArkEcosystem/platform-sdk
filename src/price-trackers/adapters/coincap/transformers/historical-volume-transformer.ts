import { KeyValuePair } from "../../../../types";
import { dayjs } from "../../../../utils/dayjs";
import { HistoricalData, HistoricalTransformer } from "../../../contracts/historical";
import { convertToCurrency } from "../utils";

export class HistoricalVolumeTransformer implements HistoricalTransformer {
	public constructor(private readonly data: KeyValuePair) {}

	public transform(options: KeyValuePair): HistoricalData {
		const { token, currency, rates, dateFormat } = options;

		const tokenId = token.toUpperCase();
		const datasets = {};

		for (const value of Object.values(this.data)) {
			datasets[dayjs(value.date).format(dateFormat)] = convertToCurrency(value.priceUsd, {
				from: currency,
				to: tokenId,
				base: tokenId,
				rates,
			});
		}

		const datasetValues: number[] = Object.values(datasets);

		return {
			labels: Object.keys(datasets),
			datasets: datasetValues,
			min: Math.min(...datasetValues),
			max: Math.max(...datasetValues),
		};
	}
}
