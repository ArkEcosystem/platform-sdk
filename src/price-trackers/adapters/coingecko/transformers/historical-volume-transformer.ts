import { KeyValuePair } from "../../../../types";
import { dayjs } from "../../../../utils/dayjs";
import { HistoricalData, HistoricalTransformer } from "../../../contracts/historical";

export class HistoricalVolumeTransformer implements HistoricalTransformer {
	public constructor(private readonly data: KeyValuePair) {}

	public transform(options: KeyValuePair): HistoricalData {
		const datasets = {};
		for (let i = 0; i < this.data.prices.length; i += 24) {
			datasets[this.data.prices[i][0]] = this.data.prices[i][1];
		}

		const datasetValues: number[] = Object.values(datasets);

		return {
			labels: Object.keys(datasets).map(time => dayjs(time).format(options.dateFormat)),
			datasets: datasetValues,
			min: Math.min(...datasetValues),
			max: Math.max(...datasetValues),
		};
	}
}
