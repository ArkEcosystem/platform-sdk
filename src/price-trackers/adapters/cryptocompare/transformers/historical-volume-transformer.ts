import { KeyValuePair } from "../../../../types";
import { dayjs } from "../../../../utils/dayjs";
import { HistoricalData, HistoricalTransformer } from "../../../contracts/historical";

export class HistoricalVolumeTransformer implements HistoricalTransformer {
	public constructor(private readonly data: KeyValuePair) {}

	public transform(options: KeyValuePair): HistoricalData {
		const datasets = this.data.map((value) => value.volumeto);

		return {
			labels: this.data.map((value) => dayjs(value.time * 1000).format(options.dateFormat)),
			datasets,
			min: Math.min(...datasets),
			max: Math.max(...datasets),
		};
	}
}
