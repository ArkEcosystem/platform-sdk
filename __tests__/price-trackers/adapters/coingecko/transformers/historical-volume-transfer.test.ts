import { HistoricalVolumeTransformer } from "../../../../../src/price-trackers/adapters/coingecko/transformers/historical-volume-transformer";

const stubResponse = require("../__fixtures__/historical-volume.json");
const stubOptions = { type: "day", dateFormat: "DD.MM" };

describe("CoinGecko", function () {
	describe("HistoricalVolumeTransformer", function () {
		it("should transform the given data", async () => {
			const subject = new HistoricalVolumeTransformer(stubResponse);

			expect(subject.transform(stubOptions)).toMatchSnapshot();
		});
	});
});
