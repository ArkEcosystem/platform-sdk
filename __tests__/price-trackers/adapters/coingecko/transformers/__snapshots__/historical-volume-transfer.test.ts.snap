// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoinGecko HistoricalVolumeTransformer should transform the given data 1`] = `
Object {
  "datasets": Array [
    4744713.640009474,
    1293301.945017782,
    995395.2750589217,
    786375.4167901358,
    2218728.193865058,
    787455.7647567509,
    524541.7432296164,
    629577.9272925577,
    796537.3487906973,
    572400.41366257,
    607816.9444040231,
    483736.2663171068,
    831681.7474437049,
    538115.8295564234,
    1009062.430593023,
    1598034.0305380435,
  ],
  "labels": Array [
    "11.12",
    "08.06",
    "13.02",
    "19.10",
    "16.12",
    "21.08",
    "28.04",
    "24.10",
    "29.06",
    "05.03",
    "01.09",
    "07.05",
    "04.11",
    "10.07",
    "17.03",
    "12.09",
  ],
  "max": 4744713.640009474,
  "min": 483736.2663171068,
}
`;
